packages/hardhat/contracts/strategy/CrabMigration.sol

```diff
@@ -148,15 +153,31 @@ import {StrategyMath} from "./base/StrategyMath.sol";
 
      /** 
       * @notice allows users to claim crabV2 shares and flash withdraw from crabV2
+      * 
+      * @param _amountToWithdraw Amount of shares to withdraw
+      * @param _maxEthToPay maximum ETH to pay to buy back the owed wSqueeth debt
       */
-     function claimAndWithdraw() external afterMigration { 
+     function claimAndWithdraw(uint256 _amountToWithdraw, uint256 _maxEthToPay) external afterMigration { 
+        uint256 amountV1Deposited = sharesDeposited[msg.sender];
+        require(_amountToWithdraw <= amountV1Deposited, "M6");
         
+        sharesDeposited[msg.sender] = amountV1Deposited - _amountToWithdraw;
```

Comment:
Wondering from the f.e perspective, what is easier to pass in?

Currently `sharesDeposited` stores the amount of V1 shares deposited. 1 V1 share isn't necessarily equal to 1 v2 share. Not fully sure why but that's what I end up getting in tests.

Do you want the `amountToWithdraw` to be in terms of v1 shares or v2 shares?  which one is easier for you to reason about from the f/e?